test/state/digital-exchange/components/actions.test.js

```diff
@@ -0,0 +1,106 @@
+
+jest.mock('api/digital-exchange/components');
+
+
+describe('state/digital-exchange/components/actions', () => {
+  let store;
+  let action;
+
+  beforeEach(() => {
+    store = mockStore(INITIAL_STATE);
+    jest.clearAllMocks();
+    store.clearActions();
+  });
+
+  describe('setDEComponents', () => {
+    it('test setDEComponents action sets the correct type', () => {
+      action = setDEComponents(LIST_DE_COMPONENTS_OK);
+      expect(action).toHaveProperty('type', SET_DE_COMPONENTS);
+    });
+  });
+
+  describe('fetchDEComponents', () => {
+    beforeEach(() => {
+      getDEComponents.mockImplementation(mockApi({ payload: LIST_DE_COMPONENTS_OK }));
+    });
+    it('fetchDEComponents calls setDEComponents and setPage actions', (done) => {
+      store.dispatch(fetchDEComponents()).then(() => {
+        const actions = store.getActions();
+        expect(actions).toHaveLength(4);
+        expect(actions[0]).toHaveProperty('type', TOGGLE_LOADING);
+        expect(actions[1]).toHaveProperty('type', SET_DE_COMPONENTS);
+        expect(actions[2]).toHaveProperty('type', TOGGLE_LOADING);
+        expect(actions[3]).toHaveProperty('type', SET_PAGE);
+        done();
+      }).catch(done.fail);
+    });
+
+    it('fetchDEComponents as error and dispatch ADD_ERRORS ', (done) => {
+      getDEComponents.mockImplementation(mockApi({ errors: true }));
+      store.dispatch(fetchDEComponents()).then(() => {
+        const actions = store.getActions();
+        expect(actions).toHaveLength(3);
+        expect(actions[0]).toHaveProperty('type', TOGGLE_LOADING);
+        expect(actions[1]).toHaveProperty('type', ADD_ERRORS);
+        expect(actions[2]).toHaveProperty('type', TOGGLE_LOADING);
+        done();
+      }).catch(done.fail);
+    });
+
+    it('components is defined and properly valued', (done) => {
+      store.dispatch(fetchDEComponents()).then(() => {
+        const actionPayload = store.getActions()[1].payload;
+        expect(actionPayload.digitalExchangeComponents).toHaveLength(4);
+        const digitalExchangeComponent = actionPayload.digitalExchangeComponents[1];
+        expect(digitalExchangeComponent).toHaveProperty('id', 'a7233e30-e6f0-4c90-9786-e3667113be12');
+        expect(digitalExchangeComponent).toHaveProperty('name', 'Avatar plugin');
+        expect(digitalExchangeComponent).toHaveProperty('lastUpdate', '2018-08-22');
+        expect(digitalExchangeComponent).toHaveProperty('marketplace', 'Entando');
+        expect(digitalExchangeComponent).toHaveProperty('version', '5.1.0');
+        expect(digitalExchangeComponent).toHaveProperty('type', 'widget');
+        expect(digitalExchangeComponent).toHaveProperty('description', 'lorem ipsum');
+        expect(digitalExchangeComponent).toHaveProperty('image', '');
+        expect(digitalExchangeComponent).toHaveProperty('rating', 3);
+        done();
+      }).catch(done.fail);
+    });
+
+    describe('test sync actions', () => {
+      describe('test s', () => {
```

Comment:
Can we provide clearer test suite and test case descriptions? For example, for a newbie like me it's not clear what 'test s' means :)